### PR TITLE
feat(*): Avoid no data panels

### PR DESCRIPTION
**Pull request by grafakus**
### ✨ Description

**Related issue(s):** solves https://github.com/grafana/explore-profiles/issues/74

This PR prevents showing panels with no data:
1. **For the grid panels**, we use different queries for the `serviceName` and the `profileMetricId`  variables, they now depend on the selected profile metric and the selected service name, respectively
2. **For the "Labels" and "Flame graph" exploration types**, the "Profile type" selector in the UI also uses a query that depends on the service selected

Short video to illustrate the changes:

https://github.com/user-attachments/assets/6e81b1c3-a173-43fe-bfa3-84816edeccbc

### 📖 Summary of the changes

We've updated `SeriesDataSource` to support multiple queries ; we took the opportunity to update the data massaging after fetching from the `/Series` endpoint, creating lookup tables for both services → profile metrics and vice-versa.

Finally, the "Hide panels without data" switch has been removed from the "All services" & "Profile types" exploration type views.

See diff tab for specific comments.

### 🧪 How to test?

- See [related issue ](https://github.com/grafana/explore-profiles/issues/74)

### CODE REVIEW

File: src/pages/ProfilesExplorerView/components/SceneByVariableRepeaterGrid/SceneByVariableRepeaterGrid.tsx [hunk 1]
```diff
@@ -74,31 +72,20 @@ export class SceneByVariableRepeaterGrid extends SceneObjectBase<SceneByVariable
     return layout === LayoutType.ROWS ? GRID_TEMPLATE_ROWS : GRID_TEMPLATE_COLUMNS;
   }
 
-  protected _variableDependency: VariableDependencyConfig<SceneByVariableRepeaterGridState> =
```
Comment (grafakus) [inline src/pages/ProfilesExplorerView/components/SceneByVariableRepeaterGrid/SceneByVariableRepeaterGrid.tsx:77]: Not needed anymore, we rely on the Scenes mechanism to update things automatically whenever a variable that is present in a query changes.

File: src/pages/ProfilesExplorerView/components/SceneExploreServiceFlameGraph/SceneExploreServiceFlameGraph.tsx [hunk 2]
```diff
@@ -11,15 +11,15 @@ import { EventViewServiceProfiles } from '../../domain/events/EventViewServicePr
...
 import { SceneFlameGraph } from './SceneFlameGraph';
 
-interface SceneExploreFlameGraphState extends SceneObjectState {
+interface SceneExploreServiceFlameGraphState extends SceneObjectState {
   mainTimeseries: SceneMainServiceTimeseries;
   body: SceneFlameGraph;
 }
 
-export class SceneExploreFlameGraph extends SceneObjectBase<SceneExploreFlameGraphState> {
+export class SceneExploreServiceFlameGraph extends SceneObjectBase<SceneExploreServiceFlameGraphState> {
```
Comment (grafakus) [inline src/pages/ProfilesExplorerView/components/SceneExploreServiceFlameGraph/SceneExploreServiceFlameGraph.tsx:19]: Naming consistency (not directly related to this PR).

**Review by simonswine — DISMISSED** [review]
LGTM. Thank you for working on this, I couldn't find any regression really.

I noticed a couple of nits, not too sure if it is worth addressing

File: src/pages/ProfilesExplorerView/infrastructure/series/SeriesDataSource.ts [hunk 3]
```diff
@@ -34,10 +32,10 @@ export class SeriesDataSource extends RuntimeDataSource {
       state: LoadingState.Done,
       data: [
         {
-          name: 'Services',
+          name: 'PyroscopeSeries',
           fields: [
             {
-              name: 'Service',
+              name: 'PyroscopeSerie',
```
Comment (bryanhuhta) [inline src/pages/ProfilesExplorerView/infrastructure/series/SeriesDataSource.ts:38]: nit: the singular of series is ...series.

english

**Review by bryanhuhta — APPROVED** [review]

File: src/pages/ProfilesExplorerView/infrastructure/series/SeriesDataSource.ts (hunk 3 above)
Comment (grafakus) [reply src/pages/ProfilesExplorerView/infrastructure/series/SeriesDataSource.ts:38]: Damn.

**Pull request merged**